### PR TITLE
Add support for `Blob`

### LINKED ISSUE #2206: Support sending Blob
### Is there an existing issue for this?

- [X] I've searched for any related issues and avoided creating a duplicate issue.

### Description

Currently, `ws` doesn't accept `Blob` as an argument to the `.send()` function:

```js
wss.on('connection', (ws) => ws.send(new Blob(['hello']))
```

I believe this is mainly for historic reasons because there used to be no `Blob` in Node.js. There is now, and it would be great to align the sending (and receiving also) with the WHATWG WebSocket standard. 

### ws version

8.16.0

### Node.js Version

v20.11.0

### System

Irrelevant.

### Expected result

`ws` supports sending `Blob` from the server.

### Actual result

- Type error on the unknown input type to `.send()`;
- A runtime `TypeError`:

```
TypeError: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Blob
```

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by lpinca**
Closes #2206

### CODE REVIEW

**Comment by lpinca** [comment]
@kettanaito I'm still not very happy with this because, other than compatibility with the WHAWG spec, it is pretty much useless. Anyway, here it is. I also didn't run any benchmarks to see the impact on performance.

File: lib/websocket.js [hunk 1]
```diff
@@ -1030,6 +1031,11 @@ function initAsClient(websocket, address, protocols, options) {
  */
 function emitErrorAndClose(websocket, err) {
   websocket._readyState = WebSocket.CLOSING;
+  //
+  // The following assignment is practically useless and is done only for
```
Comment (kettanaito) [inline lib/websocket.js:1035]: Which summarizes the entire change pretty flawlessly. I am still grateful for this consistency! Thank you!

**Review by kettanaito — APPROVED** [review]

**Review by dharishbabu007 — APPROVED** [review]

**Comment by lpinca** [comment]
@titanism are you are setting [`websocket.binaryType`](https://github.com/websockets/ws/blob/master/doc/ws.md#websocketbinarytype) to `'blob'`?

**Review by 5twelve — APPROVED** [review]

File: lib/websocket.js [hunk 2]
```diff
@@ -1188,7 +1194,10 @@ function receiverOnError(err) {
     websocket.close(err[kStatusCode]);
   }
 
-  websocket.emit('error', err);
+  if (!websocket._errorEmitted) {
+    websocket._errorEmitted = true;
+    websocket.emit('error', err);
```
Comment (nikwen) [inline lib/websocket.js:1199]: This seems worthwhile to pull out into a small function:

```js
function emitError(err) {
  websocket._errorEmitted = true;
  websocket.emit('error', err);
}
```

Seems like these lines should always be called together. Pulling them out into a function will make it less likely that someone will forget to do so in the future.

**Review by nikwen — COMMENTED** [review]
Was just looking at the commit to make sure that upgrading `ws` won't break my app. Looks great overall. Just had a small suggestion. :)

**Review by matrix-compute — APPROVED** [review]

**Pull request merged**